// suika-docs-theme/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.load = exports.ThemeContext = exports.Theme = exports.templates = exports.utils = void 0;
const Theme_1 = require("./Theme");
const fs_1 = require("fs");
const path_1 = require("path");
const THEME_NAME = "suika-theme";
const typedoc_1 = require("typedoc");
exports.utils = require("./templates/utils");
exports.templates = require("./templates");
var Theme_2 = require("./Theme");
Object.defineProperty(exports, "Theme", { enumerable: true, get: function () { return Theme_2.Theme; } });
var ThemeContext_1 = require("./ThemeContext");
Object.defineProperty(exports, "ThemeContext", { enumerable: true, get: function () { return ThemeContext_1.ThemeContext; } });
function load(app) {
    app.options.addDeclaration({
        name: "keywords",
        help: "Meta keywords",
        type: typedoc_1.ParameterType.Object,
    });
    app.options.addDeclaration({
        name: "toolbarLinks",
        help: "Toolbar navigation links",
        type: typedoc_1.ParameterType.Mixed,
    });
    app.listenTo(app.renderer, typedoc_1.RendererEvent.END, () => {
        const from = (0, path_1.resolve)(__dirname, "../dist/style.css");
        const to = (0, path_1.resolve)(app.options.getValue("out"), `assets/${THEME_NAME}.css`);
        (0, fs_1.cpSync)(from, to);
    });
    app.renderer.defineTheme(THEME_NAME, Theme_1.Theme);
}
exports.load = load;
